runtime/onert/api/src/nnfw_api_internal.cc

```diff
@@ -309,6 +312,21 @@ NNFW_STATUS nnfw_session::load_model_from_modelfile(const char *model_file_path)
     auto model = loadModel(filename, model_type);
     if (model == nullptr)
       return NNFW_STATUS_ERROR;
+
+#ifdef ONERT_TRAIN
+    constexpr const char *tinfo_name = "CIRCLE_TRAINING";
+    if (model->is_metadata_exist(tinfo_name))
+    {
+      const auto buffer = model->get_metadata(tinfo_name);
+      // parse buffer to TrainingInfo
+      _train_info = onert::traininfo_loader::loadTrainingInfo(buffer->base(), buffer->size());
+    }
```

Comment:
(note) 

Inside the `nnfw_load_model_from_modelfile`
* get data from the Model
* parse the data using `traininfo_loader::loadTrainingInfo()`